pkg/placement/selectors.go

```diff
@@ -72,3 +77,97 @@ func GetKeys(m map[string]string) []string {
 	}
 	return keys
 }
+
+// when an object is updated, we iterate over all placements and update placementToResolutionMap that
+// are affected by the update. Every affected placement decision is then queued to by synced.
+// TODO: when PlacementDecision API supports singleton statuses, this function should be updated.
+func (c *Controller) updateDecisions(obj runtime.Object) error {
+	placements, err := c.listPlacements()
+	if err != nil {
+		return err
+	}
+
+	objMR := obj.(mrObject)
+
+	changedResolutions := map[string]interface{}{}
+
+	for _, item := range placements {
+		placement, err := runtimeObjectToPlacement(item)
+		if err != nil {
+			return err
+		}
+
+		matchedSome := c.testObject(objMR, placement.Spec.Downsync)
+		if !matchedSome {
+			// if previously selected, remove
+			// TODO: optimize
+			if resolutionUpdated := c.placementResolver.RemoveObject(placement.GetName(), obj); resolutionUpdated {
+				changedResolutions[placement.GetName()] = nil
+			}
+			continue
+		}
+
+		// obj is selected by placement, update the placement decision resolver
+		resolutionUpdated, err := c.placementResolver.UpdateObjects(placement.GetName(), obj)
+		if err != nil {
+			return fmt.Errorf("failed to update placementToResolutionMap for object %v: %v",
+				util.GenerateObjectInfoString(obj), err)
+		}
+
+		if resolutionUpdated {
+			changedResolutions[placement.GetName()] = nil
+		}
+	}
+
+	// if no decision changed, we can return
+	if len(changedResolutions) == 0 {
+		return nil
+	}
+
+	// get all placement placementToResolutionMap
+	placementDecisions, err := c.listPlacementDecisions()
+	if err != nil {
+		return fmt.Errorf("failed to update placementToResolutionMap: %v", err)
+	}
+
+	handledDecisions := map[string]interface{}{}
+
+	// queue all placement placementToResolutionMap that exist are affected by the update
+	for _, pdObj := range placementDecisions {
+		name := pdObj.(mrObject).GetName()
+		if _, found := changedResolutions[name]; found {
+			// enqueue the placement decision for syncing
+			c.enqueueObject(pdObj, true)
+			// mark as handled
+			handledDecisions[name] = nil
+
+			c.logger.Info("enqueued placement decision", "name", name)
```

Comment:
The log message should say _why_ .

Reply:
I see that now there is no log message at all. When I debug controllers, I sometimes find it very helpful if I can set a `-v` level that causes logging of explanations of _why_ something was enqueued.

Reply:
To know why without adding more indicators, I changed the code to enqueue the placement-decisions right away and not build a slice of them and enqueue after.

Reply:
Thanks. The call from `func (c *Controller) handlePlacement` is not logged. Also note: if you log _after_ enqueuing, because of the lack of timing guarantees it is possible that a worker pulls the item and starts working on it (and logging about it) before the log message about enqueuing gets emitted. It would be less confusing the log _before_ enqueuing.

Reply:
Done.

Reply:
The call from `func (c *Controller) handlePlacement` is still not logged.